build/stage-build.yml

```diff
@@ -39,80 +42,100 @@
-
+  - job: OnWindows_Windows
     pool:
-      name: $(windowsPoolName)
-
+      vmImage : windows-2022
     variables:
-    - group: ${{ parameters.androidVariableGroup }}
-
-    container: windows
+      ApplicationPlatform: x64
+      ArtifactName: $(WindowsArtifactName)_${{ parameters.applicationEnvironment }}
+      ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
+      BuildTargetFramework: net6.0-windows10.0.19041.0
+    steps:
+      - template: steps-build-windows.yml
+        parameters:
+          pathToSrc: ${{ parameters.pathToSrc }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
 
+  - job: OnWindows_Tests
+    pool:
+      vmImage : windows-2022
+    variables:
+      ArtifactName: $(TestsArtifactName)_${{ parameters.applicationEnvironment }}
+      ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
     steps:
-      - template: steps-build.yml
+      - template: steps-build-tests.yml
         parameters:
           pathToSrc: ${{ parameters.pathToSrc }}
-          solutionFileName: ${{ parameters.solutionFileName }}
-          androidKeyStoreFile: ${{ parameters.androidKeyStoreFile }}
-          pathToAndroidManifest: ${{ parameters.pathToAndroidManifest }}
+          solutionFileName: '${{ parameters.solutionName }}.sln'
+          solutionName: ${{ parameters.solutionName }}
 
-  - job: On_Mac # Build iOS on a mac
-    strategy:
-      maxParallel: 3
-      matrix:
-        iOS:
-          ApplicationConfiguration: Release_iOS
-          ApplicationPlatform: Any CPU
-          ArtifactName: $(iOSArtifactName)_${{ parameters.applicationEnvironment }}
-          ApplicationEnvironment: ${{ parameters.applicationEnvironment }}
+  - job: OnWindows_Android
     pool:
-      name: $(macOSPoolName)
-      demands:
-        - Xamarin.iOS_Version -equals $(XAMARIN_IOS_VERSION)
+      vmImage : windows-2022
+    variables:
+    - name: ApplicationEnvironment
+      value: ${{ parameters.applicationEnvironment }}
+    - name: ArtifactName
+      value: $(AndroidArtifactName)_${{ parameters.applicationEnvironment }}
+    - name: ApplicationPlatform
+      value: AnyCPU
+    - name: BuildTargetFramework
+      value: net6.0-android
+
+    # For optimization, if we are on light build, we only build the project else we publish it.
+    - name: dotNetCommand
+      ${{ if eq(variables['IsLightBuild'], 'true') }}:
+        value: build
+      ${{ else }}:
+        value: publish
 
+    - group: ${{ parameters.androidVariableGroup }}
+
+    steps:
+    - template: steps-build-android.yml
+      parameters:
+        pathToSrc: ${{ parameters.pathToSrc }}
+        solutionName: ${{ parameters.solutionName }}
+        pathToAndroidManifest: ${{ parameters.pathToAndroidManifest }}
+        androidKeyStoreFile: ${{ parameters.androidKeyStoreFile }}
+
+  - job: OnMac_iOS
+    pool:
+      vmImage: macos-12
```

Comment:
are we setting a specific agent here?

Reply:
No, we only need `vmImage` to use Microsoft hosted agent.